src/components/graph/GraphCanvas.vue

```diff
@@ -307,13 +307,23 @@ watch(
       removeSlotError(node)
       const nodeErrors = lastNodeErrors?.[node.id]
       if (!nodeErrors) continue
+
+      let slotErrorsChanged = false
```

Comment:
challenge: Is there a way to do this without a `let`?

Reply:
```ts
      const validErrors = nodeErrors.errors.filter(error =>
        error.extra_info?.input_name !== undefined
      )
      const slotErrorsChanged = validErrors.length > 0 && validErrors.some(error => {
        const inputName = error.extra_info!.input_name!
        const inputIndex = node.findInputSlot(inputName)
        if (inputIndex !== -1) {
          node.inputs[inputIndex].hasErrors = true
          return true
      }
      return false
 ```
 
 i honestly don't think is is more readable.